### DISCUSSION TITLE
Restarting Production Build

### DISCUSSION THREAD

**Discussion by cvazquez** [vercel/next.js#12620] [Help]
I have a Next.js app built on my cloud VM running Ubuntu. I use the documented commands to build and run:
$ npm run build
$ npm run start

My question is, how do I restart after running a new build. Currently I have to find the pid's and issue a kill and run start again. Is there another way? I was hoping there was an npm run restart or stop, but digging into the code, I don't see that option.

**Comment by jamesmosier** [comment]
I use [pm2](https://pm2.keymetrics.io/docs/usage/quick-start/) for this very thing. I am able to restart my server with no downtime via `pm2 reload [instance_id]`. I also use an [ecosystem file](https://pm2.keymetrics.io/docs/usage/application-declaration/) to more easily manage my pm2 settings.

**Reply by cvazquez** [reply]
Thank you. That worked perfectly. 
I did notice I had to put the command in quotes.
$ sudo pm2 start "npm run start"

I assume it was confused by the second "start".

My express app I was running like so, didn't need quotes:
 $ sudo pm2 start node ./bin/www

I'm applying to my other npm scripts too.

Regards,
Carlos
